File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/table/TableRequest.java
package com.hcmute.reservation.model.dto.table;

import com.hcmute.reservation.model.enums.TableStatus;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class TableRequest {
    @NotNull @Min(1)
    private Integer capacity;

    private Boolean isActive = true;
    // Bổ sung thêm status để quản lý có thể cập nhật trạng thái thủ công (trong service)
    private TableStatus status;
    // Bổ sung version để hỗ trợ Optimistic Locking khi update (trong service)
    private Long version;
}

File: desktop-staff/reservation-winforms/DTO/table/TableRequest.cs
using Newtonsoft.Json;

namespace reservation_winforms.DTO.table
{
    public class TableRequest
    {
        [JsonProperty("capacity")]
        public int Capacity { get; set; }

        [JsonProperty("isActive")]
        public bool IsActive { get; set; } = true;

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("version")]
        public long? Version { get; set; }
    }
}